src/orderbook/state.ts

```diff
@@ -166,6 +357,26 @@ export class AccountState {
     return order
   }
 
+  /**
+   * Updates a user's order's remaining amount. For unlimited orders, the
+   * remaining amount remains unchanged.
+   *
+   * @throws If the order does not exist.
+   */
+  private updateOrderRemainingAmount(
+    user: Address,
+    orderId: number,
+    update: (amount: bigint) => bigint,
+  ) {
+    const order = this.order(user, orderId)
+    if (
+      order.priceNumerator !== UNLIMITED_ORDER_AMOUNT &&
+      order.priceDenominator !== UNLIMITED_ORDER_AMOUNT
+    ) {
```

Comment:
This is another example of logic that probably already exists somewhere in dex-contracts.

Reply:
I'm going to move these PRs to dex-contracts before continuing.